llvm/lib/CodeGen/InlineSpiller.cpp

```diff
@@ -805,6 +812,7 @@ void InlineSpiller::reMaterializeAll() {
 
     RegsToSpill[ResultPos++] = Reg;
   }
+  RegsReplaced.assign(RegsToSpill.begin() + ResultPos, RegsToSpill.end());
```

Comment:
I realize now that this can't be correct. It doesn't swap the regs into this area, it just copies the remaining ones down into the lower half. So, the rematted regs aren't preserved here.